tests/publishdependency.targets

```diff
@@ -58,4 +90,147 @@
     </Copy>
   </Target>
 
+  <Target Name="CopyNonWindowsDependecyToCoreRoot"
```

Comment:
Am i correct in assuming the targets CopyNonWindowsDependecyToCoreRoot, CopyDependecyToCoreRoot, CopyDependencyToRef all have the same logic and they just operate on different set of .csproj and project.json and ouptut directories ?

If so you could centralize the logic in a single target, parameterize the input and call them repeatedly like in https://github.com/dotnet/coreclr/blob/master/tests/runtest.proj#L308